content/blog/2023/07/22/2023-07-22-gsoc-2023-midterm.adoc

```diff
@@ -0,0 +1,183 @@
+
+== link:https://www.jenkins.io/projects/gsoc/2023/projects/add-probes-to-plugin-health-score/[Add Probes to "Plugin Health Score"]
+By link:https://github.com/Jagrutiti/[Jagruti Tiwari]
+
+A probe collects data about each plugin.
+It is an automated system ensuring that the Jenkins Community defined rules are followed.
+Probe data is used by health scoring to compute a score.
+A higher quality probe is given a good score.
+An obsolete or outdated plugin is given a lower score.
+
+Sharing details about the probes that I worked on during the first phase of GSoC.
+
+=== Unreleased Production Changes Probe
+
+A bug fix, security fix, or new features are of no use unless they are released.
+The "unreleased production changes" probe looks for unreleased commits in the production files.
+
+The probe aims to identify unreleased production commits in the Maven production files.
+
+My mentors suggested I should start with writing the test cases to reproduce the probe requirements, and later add code to pass each of the test cases.
+
+I implemented the JGit library and understood its features.
+I lacked depth in Git concepts that made me struggle with JGit.
+This led me to strengthen the Git concepts along the way.
+
+=== The Third-Party Repository Detection Probe
+
+By definition, a third-party repository is a repository that is not hosted at link:https://repo.jenkins-ci.org/[].
+
+The third-party repository detection probe ensures that plugins are built only from trusted and reliable Maven repositories.
+
+This probe was requested by the community.
+
+A major challenge in implementing this probe was considering the parent and child pom relationship from all angles.
+
+Some edge-case scenarios were not considered while specifying the probe.
+And, due to the number of open questions, this probe was put on hold until further research was completed.
+
+In the process, I have extensively read the Maven documentation to understand how to test POM structure, inheritance, and Maven API.
+
+=== GitHub Security Scan Workflow Probe
+
+The Security Scan probe identifies whether the plugin is configured to execute the security action.
+
+This probe is important from the security perspective, as it looks for known security breaches and vulnerabilities in a plugin.
+
+Its primary purpose is to verify that the security scan is properly configured in the plugin's GitHub Action.
+
+This probe is also involved in the refactoring of existing classes to avoid code duplication.
+The JEP-229 (Continuous  Delivery) probe looks for CI configurations in GitHub Actions as well.
+
+Thus, I have gained knowledge in leveraging the class hierarchy to build an abstract probe that can be implemented by any probes requiring GitHub Action configuration checks.
+
+This probe will be beneficial to the security team as it helps to identify any gaps and helps to improve the security of the Jenkins plugin ecosystem.
```

Comment:
```suggestion
The Security Scan probe identifies whether the plugin is configured to execute the security action.

From a security perspective, this probe is crucial as it checks for known security breaches and vulnerabilities in a plugin.

Its primary purpose is to ensure that the security scan is correctly configured in the plugin's GitHub Action.

Additionally, this probe involves refactoring existing classes to avoid code duplication, and it shares similarities with the JEP-229 (Continuous Delivery) probe, which also looks for CI configurations in GitHub Actions.

Throughout this process, I have gained knowledge in leveraging the class hierarchy to build an abstract probe that can be implemented by any probes requiring GitHub Action configuration checks.

The findings of this probe will be highly beneficial to the security team, as it helps identify any gaps and enables the improvement of the overall security of the Jenkins plugin ecosystem.
```